docker-fzf

```diff
@@ -339,13 +339,147 @@ dkrma() {
+
+    #column1: comma seperated list of ids to be removed if this option is selected
+    #column2: text to be displayed to the user.
+    #seperator: tab ("\t")
+    local images='';
+
+    # create image references for non-dangling images
+    for id in $(docker images --format "{{.ID}}" --filter "dangling=false" | sort | uniq); do #remove duplicate ids. necessary because each image tag has an entry, and images may have multiple tags
+      local references=$(docker inspect "$id" --format "{{range .RepoTags}}{{.}}\n{{end}}{{range .RepoDigests}}{{.}}\n{{end}}");
+
+      local list=$(echo "$references" | sed 's/\\n$//' | sed -r 's/\\n/\n/g'); #remove trailing new line && replace \n with actual new line
+      local listLength=$(echo "$list" | wc -l);
+      local head=$(echo "$list" | head -n 1);
+      local tail=$(echo "$list" | tail -n 1);
+
+      local result="$id\t";
+      for ref in $(echo -e "$list"); do
+        if [[ "$head" == "$ref" ]]; then
+          result+="$ref";
+          if [[ "$listLength" == "2" ]]; then
+            result+=" (alias: ";
+          elif [[ "$listLength" != "1" ]]; then
+            result+=" (aliases: ";
+          fi;
+        elif [[ "$tail" == "$ref" ]]; then
+            result+="$ref)";
+        else
+            result+="$ref, ";
+        fi;
+      done;
+
+      if [[ "$images" == "" ]]; then
+        images="$result";
+      else
+        images+="\n$result";
+      fi
+    done;
+
+  # create image references for dangling images
+    local ids=$(docker images --format "{{.ID}}" --filter "dangling=true");
+    local numberOfIds=$(echo "$ids" | wc -l)
+    if [[ "$numberOfIds" != "0" ]]; then
+      #add option to remove all dangling images
+      local allDanglingImages=$(echo -e "$ids" | sed ':a; N; s/\n/,/; ta'); # join all ids into CSV list
+      allDanglingImages+="\tAll dangling images (total of $numberOfIds)";
+      if [[ "$images" == "" ]]; then
+        images="$allDanglingImages";
+      else
+        images+="\n$allDanglingImages";
+      fi
+
+      #add options, group by label
+      local idCreatedLabel='';
+      for id in $(echo "$ids" | head -n 20); do
+        local created=$(docker inspect "$id" --format "{{.Created}}");
+        local labels=$(docker inspect "$id" --format '{{ range $key, $value := .Config.Labels }}{{ $key }}:{{$value}}\n{{end}}');
+
+        while read -r label; do #also entered for no label
+          local reference=$(echo -e "$id\t$created\t$label");
+          if [[ "$idCreatedLabel" == "" ]]; then
+            idCreatedLabel="$reference";
+          else
+            idCreatedLabel+="\n$reference";
+          fi
+        done <<< "$(echo "$labels" | sed 's/\\n$//' | sed -r 's/\\n/\n/g')"; #remove trailing new line && replace \n with actual new line
+      done;
+
+      local awk=''
+      #$1=id
+      #$2=created date
+      #$3=label
+      #a = all ids that belong to label
+      #b = max created date
+      #c = max created id
+      awk+='{'
```

Comment:
Hm, any reason you can't just write a multi-line script here?

```sh
awk = '
a[$3]=a[$3] ? a[$3]","$1 : $1'
...
```

Reply:
wasn't aware that this is possible, changing it right away!